src/platform/utilities/sso/index.js

```diff
@@ -44,16 +44,17 @@ export async function ssoKeepAliveSession() {
 }
 
 export async function checkAutoSession(application = null, to = null) {
-  const [{ ttl, authn }, hasSession] = await Promise.all([
+  const [{ ttl, authn }, userProfile] = await Promise.all([
     ssoKeepAliveSession(),
-    hasVAGovSession(),
+    vaGovProfile(),
   ]);
-  if (hasSession && ttl === 0) {
+  // FIXME: will a 3rd party lib auto convert the "sign_in" snake case to camel?
+  if (userProfile?.sign_in?.ssoe && ttl === 0) {
     // explicitly check to see if the TTL for the SSO3 session is 0, as it
     // could also be null if we failed to get a response from the SSOe server,
     // in which case we don't want to logout the user because we don't know
     logout('v1', 'sso-automatic-logout');
-  } else if (!hasSession && ttl > 0 && !getLoginAttempted() && authn) {
+  } else if (!userProfile && ttl > 0 && !getLoginAttempted() && authn) {
```

Comment:
it looks like you're returning an empty object in the non-ssoe profile calls. unfortunately this is still a truthy value for javascript, so probably should just return `null` above

Reply:
good catch!